### PR TITLE
Fix some typed storage is deprecated warnings.

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #89867

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

cc @zou3519 @Chillee @samdow @soumith

### CODE REVIEW

**Review by albanD — APPROVED** [review]
SGTM
If you want to test this, you can do the same as this test: https://github.com/pytorch/pytorch/blob/a029ec2c88edf430e682789684f1b27bfe607bf2/test/test_mps.py#L7171-L7211

File: functorch/_src/aot_autograd.py [hunk 1]
```diff
@@ -399,7 +399,7 @@ def filter_and_record_aliased_outs(outputs):
             # This will be more complicated when you have multiple _base tensors aliasing the same
             # underlying storage, when we eventually handle that.
             # We'll need to ensure that we generate the view off of the right base.
-            inp_storage_refs = {StorageWeakRef(inpt.storage()): idx for idx, inpt in enumerate(flat_f_args)}
+            inp_storage_refs = {StorageWeakRef(inpt._storage()): idx for idx, inpt in enumerate(flat_f_args)}
```
Comment (kurtamohler) [inline functorch/_src/aot_autograd.py:402]: Just double checking that you remember that `Tensor._storage()` gives an `UntypedStorage`, whereas `Tensor.storage()` gave a `TypedStorage`. But if these changes aren't causing any failures, then that's probably okay

Comment (ezyang) [reply functorch/_src/aot_autograd.py:402]: yes, I want untyped here

**Pull request closed without merging**